src/framework/constraints.ml

```diff
@@ -258,6 +258,34 @@ struct
         let ad = man.ask (Queries.EvalFunvar e) in
         Queries.AD.to_var_may ad (* TODO: don't convert, handle UnknownPtr below *)
     in
+    let once once_control init_routine =
```

Comment:
This begs my expected question: could this also be a `Spec` lifter? I haven't looked at this deeply enough to know whether this should be actually doable.